frontend/packages/dev-console/src/components/import/ImportStrategySelector.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { FormGroup, Grid, GridItem, Tile } from '@patternfly/react-core';
+import { LayerGroupIcon, CubeIcon, GitAltIcon, StarIcon } from '@patternfly/react-icons';
+import { FormikValues, useField, useFormikContext } from 'formik';
+import { useTranslation } from 'react-i18next';
+import { ImportStrategy } from '@console/git-service/src';
+import { getFieldId } from '@console/shared/src';
+import { DetectedStrategyFormData } from './import-types';
+import './ImportStrategySelector.scss';
+
+const ImportStrategySelector: React.FC = () => {
+  const { t } = useTranslation();
+  const {
+    values: {
+      import: { recommendedStrategy },
+    },
+    setFieldValue,
+    setFieldTouched,
+  } = useFormikContext<FormikValues>();
+  const [selected] = useField<DetectedStrategyFormData>('import.selectedStrategy');
+
+  const fieldId = getFieldId('import.selectedStrategy', 'importStrategySelect');
+
+  const itemList = [
+    {
+      name: 'Devfile',
+      type: ImportStrategy.DEVFILE,
+      priority: 2,
+      detectedFiles: [],
+      icon: <LayerGroupIcon />,
+    },
+    {
+      name: 'Dockerfile',
+      type: ImportStrategy.DOCKERFILE,
+      priority: 1,
+      detectedFiles: [],
+      icon: <CubeIcon />,
+    },
+    {
+      name: 'Builder Image',
+      type: ImportStrategy.S2I,
+      priority: 0,
+      detectedFiles: [],
+      icon: <GitAltIcon />,
+    },
+  ];
+
+  const onSelect = React.useCallback(
+    (item) => {
+      setFieldValue('import.selectedStrategy.name', item.name);
+      setFieldValue('import.selectedStrategy.type', item.type);
+      setFieldValue('import.selectedStrategy.priority', item.priority);
+      setFieldValue('import.selectedStrategy.detectedFiles', item.detectedFiles);
+      setFieldTouched('import.selectedStrategy', true);
```

Comment:
You would also need to update the `build.strategy` here to use proper submit utils for each import strategy.